Failed to run tests on Mac OS M1

I run build and then test a case, but it seems that are some link errors.

My commands
```
cargo build --target=aarch64-apple-darwin
cargo test --color=always --package lance --lib dataset::scanner::test::test_knn_nodes --target=aarch64-apple-darwin
```

Error Infos

  = note: Undefined symbols for architecture arm64:
            "_dot_f16", referenced from:
                _$LT$arrow_array..types..Float16Type$u20$as$u20$lance_linalg..distance..dot..Dot$GT$::dot::ha40402edb942cfc7 in liblance_linalg-918df73e4e3312fb.rlib(lance_linalg-918df73e4e3312fb.134ejned65ua16fy.rcgu.o)
            "_l2_f16", referenced from:
                _$LT$arrow_array..types..Float16Type$u20$as$u20$lance_linalg..distance..l2..L2$GT$::l2::h03acdb62496bef35 in liblance_index-4f8d82acf829c66f.rlib(lance_index-4f8d82acf829c66f.6kt4alzxtvde7ta.rcgu.o)
                _$LT$arrow_array..types..Float16Type$u20$as$u20$lance_linalg..distance..l2..L2$GT$::l2::h357abd8dfe513687 in liblance_linalg-918df73e4e3312fb.rlib(lance_linalg-918df73e4e3312fb.134ejned65ua16fy.rcgu.o)
          ld: symbol(s) not found for architecture arm64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: could not compile `lance` (lib test) due to 1 previous error

--- a/rust/lance-index/benches/hnsw.rs
+++ b/rust/lance-index/benches/hnsw.rs
@@ -29,7 +29,7 @@ use lance_testing::datagen::generate_random_array_with_seed;
 
 fn bench_hnsw(c: &mut Criterion) {
     const DIMENSION: usize = 1024;
-    const TOTAL: usize = 1 * 1024 * 1024;
+    const TOTAL: usize = 1024 * 1024;
     const SEED: [u8; 32] = [42; 32];
     const K: usize = 10;
 

--- a/rust/lance-linalg/build.rs
+++ b/rust/lance-linalg/build.rs
@@ -1,4 +1,26 @@
+// Copyright 2024 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::env;
+
 fn main() {
+    let rust_toolchain = env::var("RUSTUP_TOOLCHAIN").unwrap();
+    if rust_toolchain.starts_with("nightly") {
+        // enable the 'nightly' feature flag
+        println!("cargo:rustc-cfg=feature=\"nightly\"");
+    }
+
     println!("cargo:rerun-if-changed=src/simd/f16.c");
 
     if cfg!(all(target_os = "macos", target_feature = "neon")) {
@@ -6,7 +28,6 @@ fn main() {
             .compiler("clang")
             .file("src/simd/f16.c")
             .flag("-mtune=apple-m1")
-            .flag("-ffast-math")
             .flag("-O3")
             .flag("-Wall")
             .flag("-Werror")
@@ -22,7 +43,6 @@ fn main() {
             .std("c17")
             .file("src/simd/f16.c")
             .flag("-march=sapphirerapids")
-            .flag("-ffast-math")
             .flag("-mavx512f")
             .flag("-mavx512vl")
             .flag("-mavx512bw")

--- a/rust/lance-linalg/build.rs
+++ b/rust/lance-linalg/build.rs
@@ -23,7 +23,7 @@ fn main() {
 
     println!("cargo:rerun-if-changed=src/simd/f16.c");
 
-    if cfg!(all(target_os = "macos", target_feature = "neon")) {
+    if cfg!(all(target_os = "macos", target_arch = "aarch64")) {
         cc::Build::new()
             .compiler("clang")
             .file("src/simd/f16.c")

